index.rst

```diff
@@ -1,59 +1,367 @@
+
+ - All production operators and science pipelines developers have access to the ``auxiliaries`` role, which provides write access to the ``refcats`` and ``skymap`` directores.  This role is used to create per-ticket subdirectories in ``refcats`` prior to starting work on ingesting a new reference catalog, and used directly to run ``butler register-skymap``.
+
+In addition to these filesystem-level controls, we also plan to provide some *informal* protections based on the the same roles in the butler client: the ``Butler`` and ``Registry`` classes (and associated command-line tools) will accept a ``role`` argument that permits write operations on collections with certain associated prefixes.
+The default role is the user's unix username, which provides write access only to ``u/<user>`` collections.
+These guards against careless fingers, not careless brains - we will not attempt to restrict which roles a user can assume *at the butler client level*.
+
+.. note::
+
+   This proposal intentionally makes no mention of the RFC process that is currently in place for ``/datasets`` for Gen2, or the ``/project`` filesystem.
+
+   In practice, RFCs for most modifications - certainly routine ingests or calibration updates - almost never exceed our `Empowerment of DM team members`_ criteria, and asking for sysadmins who are not domain experts to do the actual work both increases friction and increases the chance something will go wrong (or, rather, the chances that if something does go wrong, it is not fixed immediately).
+
+   This proposal makes no objection to having a separate ``/project`` filesystem for files.
+   But using ``/project`` and symlinks as a way to work around restrictions on write access to appropriate subsets of ``/datasets`` is just that - a workaround - and one that makes it more difficult than it ought to be to find things.
+   If filesystem-level controls (or quotas, etc.) really are necessary even for ``/datasets`` (note that we already allow for filesystem-level controls by symlinking raws from ``/lsstdata`` and ``/external-raw``), I think this is something we can tolerate, but regular directory permission controls within ``/datasets`` would be preferable.
+   If we do use symlinks from ``/project``, we should at least ensure that a::
+
+       /datasets/repo/u/<user> -> /project/<user>/.datarepo
+
+   link (or equivalent) is automatically created with the right permissions for all users, and encourage access only via that symlink (hence the hidden directory as a target).
```

Comment:
I have no objection to this, but this (automatic?) symlink between e.g. `/datasets` and `/project` needs to be clearly explained in the developer guide, because its utility and the fact that I shouldn't touch it on disk is not inherently obvious.